src/main/validation/PromotionQueryValidation.ts

```diff
@@ -27,6 +28,17 @@ export class PromotionQueryValidation {
     /** note incoming format is a string, and Joi will automatically convert to type Date */
     expirationDate: Joi.date(),
     dayOfWeek: Joi.string().valid(...Object.values(Day)),
+    sort: Joi.string().valid(...Object.values(SortOptions)),
+    lat: Joi.any().when('sort', {
+      is: SortOptions.DISTANCE,
+      then: Joi.number().required(),
+      otherwise: Joi.number(),
+    }),
```

Comment:
I found out Joi has this cool "if then" syntax. I've made it so that if we've specified sort by distance, then we need to make sure lat and lon are both defined

Reply:
Oooh this is actually really cool, glad you found this!!!